s4e-backend/src/test/java/pl/cyfronet/s4e/admin/scene/AdminSceneControllerTest.java

```diff
@@ -0,0 +1,323 @@
+package pl.cyfronet.s4e.admin.scene;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.test.web.servlet.MockMvc;
+import pl.cyfronet.s4e.*;
+import pl.cyfronet.s4e.bean.*;
+import pl.cyfronet.s4e.data.repository.AppUserRepository;
+import pl.cyfronet.s4e.data.repository.ProductRepository;
+import pl.cyfronet.s4e.data.repository.SceneRepository;
+import pl.cyfronet.s4e.data.repository.SchemaRepository;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static pl.cyfronet.s4e.TestJwtUtil.jwtBearerToken;
+
+@BasicTest
+@AutoConfigureMockMvc
+public class AdminSceneControllerTest {
+    private static final String SCHEMA_PATH_PREFIX = "classpath:schema/";
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private SchemaRepository schemaRepository;
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Autowired
+    private SceneRepository sceneRepository;
+
+    @Autowired
+    private AppUserRepository appUserRepository;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private TestDbHelper testDbHelper;
+
+    @Autowired
+    private TestResourceHelper testResourceHelper;
+
+    private AppUser admin;
+    private AppUser user;
+
+    private Map<String, Schema> schemas;
+    private Map<String, Product> products;
+    private Map<String, List<Scene>> productScenes;
+    private long maxSceneId;
+
+    @BeforeEach
+    public void beforeEach() {
+        testDbHelper.clean();
+
+        admin = appUserRepository.save(AppUser.builder()
```

Comment:
Could you move this builder logic to `UserFactory`?

Reply:
I think it's in scope of #637.